test/tests/classTransformer.test.ts

```diff
@@ -0,0 +1,118 @@
+import { ObjectId } from 'bson';
+import { classToPlain, Exclude, Expose, plainToClass, Transform } from 'class-transformer';
+
+import { prop } from '../../src/prop';
+import { getModelForClass } from '../../src/typegoose';
+
+// re-implement base Document to allow class-transformer to serialize/deserialize its properties
+class Document {
+  @Expose()
+  @Transform(
+    // makes sure that when deserializing from a Mongoose Object, ObjectId is serialized into a string 
+    (value: any) => value instanceof ObjectId
+    ? value.toHexString()
+    : value,
+    { toClassOnly: true })
+  public _id: string
+  
+  @Expose()
+  public __v: number
+}
+
+@Exclude()
+class Account extends Document {
+  @prop()
+  @Expose()
+  public email: string
+
+  @prop()
+  @Expose({ groups: ['admin'] })
+  password: string;
+}
+
+const AccountModel = getModelForClass(Account)
+
+describe('class-transformer', () => {
+  let id: string
+  beforeAll(async () => {
+    const { _id } = await AccountModel.create({
+      email: 'somebody@gmail.com',
+      password: 'secret',
+    } as Account)
+    // note here that _id is an ObjectId, hence the toString()
+    // otherwise it will have the shape of : { _bsonType: 'ObjectId', id: ArrayBuffer }
+    id = _id.toString()
+  })
+  describe('lean', () => {
+    it('should be able to use class-transformer', async () => {
+      // lean return a Plain Old Javascript Object
+      const POJO = await AccountModel.findById(id).lean()
```

Comment:
```suggestion
      const POJO = await AccountModel.findById(id).orFail().lean()
```

to remove the `| undefined` from the type and ensure the test fails if the id cannot be found

Reply:
Yes sure, currently doing it.